### PR TITLE
Added clipping, fbo invalidate, shader api, boolean state query and s…

### PR DESCRIPTION
**Pull request by jasomobica**
…ync tests.

Plus cleanup of HTML files.

Signed-off-by: Janusz Sobczak <janusz.sobczak@mobica.com>

### CODE REVIEW

**Comment by toji** [comment]
I'd like to requests that simple, noisy changes like the removal of the DOCTYPE here be done in their own pull request. The way that Github presents changes for review makes it hard to filter those simple changes out and focus on the newly added tests.

**Comment by jasomobica** [comment]
Sorry about that. I am traditionally squashing commits before creating pull requests.

File: sdk/tests/deqp/modules/shared/glsStateQuery.js [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+/**
+ * Verify that WebGL state 'param' has the expected value
+ * @param {number} param
+ * @param {number|Array<number>} reference
+ * @return {boolean}
+ */
+glsStateQuery.verifyGreaterOrEqual = function(param, reference) {
+    var value = gl.getParameter(param);
+    if (reference instanceof Array) {
```
Comment (kenrussell) [inline sdk/tests/deqp/modules/shared/glsStateQuery.js]: Not a big deal, but perhaps a glsStateQuery.compareImpl could be added which takes a function object as argument which is used to compare each individual array element in the case of arrays. That way this and verifyLessOrEqual could have less duplicated code.

**Comment by kenrussell** [comment]
Thanks for this contribution. Going forward I agree with Brandon's assessment that large, boilerplate changes like the doctype removal here should be done in independent pull requests. Per our offline discussion I realize that this would cause a bit of a headache with this pull request, so merging this as is now. Other than that, this looks excellent.

**Pull request merged**